src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,150 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    private static class Node<T> {
+        T data;
+        Node<T> next;
+        Node<T> prev;
```

Comment:
Don't use getters and setters in class Node. We have access to private fields of the inner class, so using getters or setters is redundant. Make these fields private and access them directly within the outer class.